File: mudu_api/csharp/uni/UniProcedureParam.cs
namespace Universal {

using MessagePack;
using MessagePack.Formatters;
using System.Collections.Generic;




[MessagePackObject]
public struct UniProcedureParam {
    
    [global::System.Diagnostics.CodeAnalysis.SetsRequiredMembers]
    public UniProcedureParam()
    {
        
        Procedure = 0;
        
        Session = new UniOid();
        
        ParamList = [];
        
    }
    
    
    
    [Key(0)]
    public ulong Procedure { get; set; }
    
    
    [Key(1)]
    public required UniOid Session { get; set; }
    
    
    [Key(2)]
    public required List<UniDatValue> ParamList { get; set; }
    
}

}

File: mudu_binding/src/universal/uni_procedure_param.rs
use crate::universal::uni_oid::UniOid;

use crate::universal::uni_dat_value::UniDatValue;

#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, Default)]
pub struct UniProcedureParam {
    pub procedure: u64,

    pub session: UniOid,

    pub param_list: Vec<UniDatValue>,
}
